extensions/BUILD.gn

```diff
@@ -15,9 +13,14 @@ if (is_android || is_ios) {
       "//base",
       "//components/url_pattern_index",
       "//extensions/buildflags",
+      "//net:net",
       "//url",
     ]
 
+    if (is_android) {
```

Comment:
@darkdh, you added `|| is_ios` at the top here in https://github.com/brave/brave-core/pull/5832/files?file-filters%5B%5D=#diff-fcc0f39b7726fc6e45509d18d51695deR1, but @Gyuyoung now had to add a dep guard here for android only because `ERROR at //content/BUILD.gn:18:1: Assertion failed.
assert(!is_ios, "Chromium/iOS shouldn't use anything in //content")`. I am not sure why gn check doesn't fail for iOS here, but `//extensions/common/url_pattern.cc` includes `content/public/common/url_constants.h` without any guards so it would seem incorrect to not add the dep. Does iOS even use this?

Reply:
I'm not entirely sure. I was asked to add that during the review. I think that is for `//brave/common:shield_exceptions` without adding extension guards which will break android. @bridiver can explain this trick better

Reply:
I asked if iOS CI checks gn check on slack. @linhkikuchi answered the iOS CI checks it. If so, I start to wonder if iOS builds these files.

Reply:
the purpose of this target is to make a subset of extension sources available to android because (unfortunately) some of our code uses extension/url_pattern instead of components/url_pattern. I believe `is_ios` here https://github.com/brave/brave-core/pull/6568/files/5fd767cb70ef49aa29e3571c9a02fa9c14a9b419#diff-fcc0f39b7726fc6e45509d18d51695deR1 is incorrect. You cannot make `extensions/common/url_pattern.cc` available to ios if it has an include from content and you can't just leave the dep out when there's an include for it.

Reply:
so to be clear, the fix is removing `is_ios` here https://github.com/brave/brave-core/pull/6568/files/5fd767cb70ef49aa29e3571c9a02fa9c14a9b419#diff-fcc0f39b7726fc6e45509d18d51695deR1